hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1240 @@
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    adults = fields.Integer(
+        "Adults",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="List of adults there in guest list. ",
+    )
+    children = fields.Integer(
+        "Children",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="Number of children there in guest list.",
+    )
+    reservation_line = fields.One2many(
+        "hotel_reservation.line",
+        "line_id",
+        "Reservation Line",
+        help="Hotel room reservation details.",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("confirm", "Confirm"),
+            ("cancel", "Cancel"),
+            ("done", "Done"),
+        ],
+        "State",
+        readonly=True,
+        default=lambda *a: "draft",
+    )
+    folio_id = fields.Many2many(
+        "hotel.folio",
+        "hotel_folio_reservation_rel",
+        "order_id",
+        "invoice_id",
+        string="Folio",
+    )
+    no_of_folio = fields.Integer("No. Folio", compute="_compute_folio_id")
+
+    def _compute_folio_id(self):
+        folio_list = []
+        for res in self:
+            for folio in res.folio_id:
+                folio_list.append(folio.id)
+            folio_len = len(folio_list)
+            res.no_of_folio = folio_len
+        return folio_len
+
+    def unlink(self):
+        """
+        Overrides orm unlink method.
+        @param self: The object pointer
+        @return: True/False.
+        """
+        for reserv_rec in self:
+            if reserv_rec.state != "draft":
+                raise ValidationError(
+                    _(
+                        "Sorry, you can only delete the reservation when it's draft!"
+                    )
+                )
+        return super(HotelReservation, self).unlink()
+
+    def copy(self):
+        ctx = dict(self._context) or {}
+        ctx.update({"duplicate": True})
+        return super(HotelReservation, self.with_context(ctx)).copy()
+
+    @api.constrains("reservation_line", "adults", "children")
+    def check_reservation_rooms(self):
+        """
+        This method is used to validate the reservation_line.
+        -----------------------------------------------------
+        @param self: object pointer
+        @return: raise a warning depending on the validation
+        """
+        ctx = dict(self._context) or {}
+        for reservation in self:
+            cap = 0
+            for rec in reservation.reservation_line:
+                if len(rec.reserve) == 0:
+                    raise ValidationError(
+                        _("Please Select Rooms For Reservation.")
+                    )
+                for room in rec.reserve:
+                    cap += room.capacity
```

Comment:
In single line code 
cap = sum(room.capacity for room in rec.reserve)

Reply:
Done